File: frontEnd/train_truck/lib/Models/Route.dart
import 'package:train_truck/Models/Trip.dart';

class Route {
  int? routeId;
  int? agancy;
  String? routeShortName=" ";
  String? routeLongName=" ";
  List<Trip>? trip;
  String? etat;


  Route(
      {this.routeId,
         this.agancy,
        this.trip,
        this.routeShortName,
         this.routeLongName,
      this.etat});

  Route.fromJson(Map<String, dynamic> json) {
    routeId = json['route_id'];
    agancy = json['agancy'];
    routeShortName = json['routeShortName'];
    routeLongName = json['routeLongName'];
    etat=json["etat"];

    if (json['trips'] != null) {
      trip = <Trip>[];
      json['trips'].forEach((v) {
        trip!.add(new Trip.fromJson(v));
      });
    }
  }

  Map<String, dynamic> toJson() {
    final Map<String, dynamic> data = new Map<String, dynamic>();
    data['route_id'] = this.routeId;
    data['agancy'] = this.agancy;
    data['route_short_name'] = this.routeShortName;
    data['route_long_name'] = this.routeLongName;
    return data;
  }
}

File: backend/trainTruck/src/main/java/com/trainTruck/demo/Model/Route.java
package com.trainTruck.demo.Model;

import java.util.List;

import org.hibernate.annotations.ColumnTransformer;

import com.fasterxml.jackson.annotation.JsonIgnoreProperties;

import jakarta.persistence.CascadeType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.OneToMany;
import jakarta.persistence.Table;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Setter
@Getter
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Entity
@Table(name="routes")
public class Route {
	@Id
	@GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "route_id")
	private int route_id;
	@Column(name="agancy_id")
	private int agancy;
	@JsonIgnoreProperties("route_id")
	@OneToMany(mappedBy = "route_id")
	private List<Trip> trips;
	
	@Column(name="route_short_name")
	private String routeShortName;
	
	
	@Column(name ="route_long_name" )
	private String routeLongName;
	
	private String route_desc;
	private String route_url;
	public String getRouteLongName()
	{
		return routeLongName.toLowerCase();
	}
	
	
	
	
	
}
